python/test/test_dataset.py

```diff
@@ -182,6 +182,34 @@ def test_view_subdata(self):
         self.assertEqual(view.dimensions[Dim.Z], 4)
         self.assertEqual(len(view), 4)
 
+    def test_insert_subdata(self):
+        d1 = Dataset()
+        d1[Data.Value, "a"] = ([Dim.X], np.arange(10, dtype="double"))
```

Comment:
Should add coordinates in these tests, to make sure this behaves as intended. The insert should succeed if coordinates match, but fail if there is a mismatch,

Reply:
I can see you added a coordinate in the `cpp` test. Do you also want to add one in the python?

Reply:
@nvaytet good suggestion